numpy/random/tests/test_generator_mt19937.py

```diff
@@ -729,6 +729,21 @@ def test_shuffle(self):
             desired = conv([4, 1, 9, 8, 0, 5, 3, 6, 2, 7])
             assert_array_equal(actual, desired)
 
+        # Test shuffle on multidimensional array along a different axis
```

Comment:
The tests immediately above this cover a variety of input types. Should the `axis` argument be paired in combination with them?

Reply:
I am not sure. If combine them, we might need to test errors for different input types. For example `lambda x: np.array([])` which has one dimension, if we set `axis=1`, do we need to assert `AxisError`?

Reply:
Yes, `axis=1` should raise `AxisError` on a 1d array

Reply:
In that case, will it be better to combine them since we need to raise errors for some but not all types?